cpp/roaring.hh

```diff
@@ -489,26 +537,28 @@ public:
   }
 
   bool operator==(const RoaringSetBitForwardIterator &o) {
-    return i.current_value == *o;
+    return i.current_value == *o && i.has_value == o.i.has_value;
   }
 
   bool operator!=(const RoaringSetBitForwardIterator &o) {
-    return i.current_value != *o;
+    return i.current_value != *o || i.has_value != o.i.has_value;
   }
 
   RoaringSetBitForwardIterator(const Roaring & parent, bool exhausted = false) {
     if(exhausted) {
-        i.parent = parent.roaring;
+        i.parent = &parent.roaring;
         i.container_index = INT32_MAX;
         i.has_value = false;
         i.current_value = UINT32_MAX;
     } else {
-      roaring_init_iterator(parent.roaring, &i);
+      roaring_init_iterator(&parent.roaring, &i);
     }
   }
 
-  virtual ~RoaringSetBitForwardIterator() {
```

Comment:
Virtual here only helps the case where a user subclasses this object and uses it in a polymorphic way (including deleting it from a polymorphic context). I would argue that it isn't the job of this library to worry about this case and that it is very unlikely that a user would want to do that regardless.